File: src/#1448/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var goodNodes = function (root) {
  let count = 0

  function helper(root, maxNum) {
    if (!root) {
      return
    }
    if (root.val >= maxNum) {
      count++
      maxNum = root.val
    }
    helper(root.left, maxNum)
    helper(root.right, maxNum)
  }
  helper(root, -10000)
  return count
}

File: src/#1448/solution.go
package solution

type TreeNode struct {
	Val         int
	Left, Right *TreeNode
}

func goodNodes(root *TreeNode) int {
	count := 0
	helper(root, -10000, &count)
	return count

}
func helper(root *TreeNode, maxNum int, count *int) {
	if root == nil {
		return
	}
	if root.Val >= maxNum {
		*count += 1
		maxNum = root.Val
	}
	helper(root.Left, maxNum, count)
	helper(root.Right, maxNum, count)
}
